utils/dev-tool/src/main.rs

```diff
@@ -0,0 +1,104 @@
+mod android;
+mod apple;
+mod server;
+mod setup;
+mod utils;
+mod workspace;
+
+use std::path::PathBuf;
+use std::{env, fs};
+use structopt::StructOpt;
+
+#[derive(Debug, PartialEq, StructOpt)]
+#[structopt(about = "Lockbook's development and ci tool.")]
+enum Commands {
+    // CI steps in order --------------
+    /// Check the formatting of the workspace
+    CheckWorkspaceFmt,
+
+    /// Check the lint of the workspace
+    CheckWorkspaceClippy,
+
+    /// Run the server detached
+    RunServer,
+
+    /// Run all rust tests
+    RunRustTests,
+
+    /// Check the formatting of the android client
+    CheckAndroidFmt,
+
+    /// Check the lint of the android client
+    CheckAndroidLint,
+
+    /// Run kotlin integration tests
+    RunKotlinTests,
+
+    /// Run the swift integration tests
+    RunSwiftTests,
+
+    /// Kill the server for commit hash
+    KillServer,
+
+    // End of CI steps --------------
+    /// Verify CI environment
+    VerifyCIEnvironment,
+
+    /// Make kotlin jni libs
+    MakeKotlinLibs,
+
+    /// Make kotlin jni libs for tests
+    MakeKotlinTestLib,
+
+    /// Make swift jni libs for tests
+    MakeSwiftTestLib,
+
+    /// Kill all servers running
+    KillAllServers,
+}
+
+pub struct ToolEnvironment {
+    root_dir: PathBuf,
+    target_dir: PathBuf,
+    hash_info_dir: PathBuf,
+    commit_hash: String,
+}
+
+impl Default for ToolEnvironment {
+    fn default() -> Self {
```

Comment:
This function makes file system calls, so this isn't a good implementation of the `Default` trait. It's also unnecessary because you are just using it to initialize `ToolEnvironment` in the first line in main.

I propose just moving this code into main.

However, if for some reason you wanted to keep `Default` around, then just use it to set the default values and then place the file system / init logic somewhere else. But really I see no need for this.

Reply:
I agree about the default part being weird, unsure whether the implementation detail needs to end up in main

Reply:
I mean the body of the `default` function as it is now can basically just be moved into `main`

Reply:
inlined into main :heavy_check_mark: